samplers/aws/xray/remote_sampler.go

```diff
@@ -0,0 +1,219 @@
+	}
+
+	remoteSampler := &remoteSampler{
+		clock:                        clock,
+		manifest:                     m,
+		clientID:                     id,
+		xrayClient:                   newClient(cfg.endpoint),
+		samplingRulesPollingInterval: cfg.samplingRulesPollingInterval,
+	}
+
+	// starts the rule poller
+	remoteSampler.start(ctx)
+
+	return remoteSampler, nil
+}
+
+func (rs *remoteSampler) ShouldSample(parameters sdktrace.SamplingParameters) sdktrace.SamplingResult {
+	// ToDo: add business logic for remote sampling
+
+	return sdktrace.TraceIDRatioBased(0.05).ShouldSample(parameters)
+}
+
+func (rs *remoteSampler) Description() string {
+	return "remote sampling with AWS X-Ray"
+}
+
+func (rs *remoteSampler) start(ctx context.Context) {
+	if !rs.pollerStart {
+		rs.pollerStart = true
+		rs.startRulePoller(ctx)
+	}
+}
+
+func (rs *remoteSampler) startRulePoller(ctx context.Context) {
+	go func() {
+		// Period = 300s, Jitter = 5s
+		t := newTicker(rs.samplingRulesPollingInterval, 5*time.Second)
+
+		// Periodic manifest refresh
+		for {
+			if err := rs.refreshManifest(ctx); err != nil {
+				globalLogger.Error(err, "Error occurred while refreshing sampling rules")
+			} else {
+				globalLogger.Info("Successfully fetched sampling rules")
+			}
+			select {
+			case _, more := <-t.C():
+				if !more {
+					return
+				}
+				continue
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+}
+
+func (rs *remoteSampler) refreshManifest(ctx context.Context) (err error) {
+	// Compute 'now' before calling GetSamplingRules to avoid marking manifest as
+	// fresher than it actually is.
+	now := rs.clock.now().Unix()
+
+	// Get sampling rules from proxy.
+	rules, err := rs.xrayClient.getSamplingRules(ctx)
+	if err != nil {
+		return
+	}
+
+	failed := false
+
+	// temporary manifest declaration.
+	tempManifest := &manifest{
+		rules: []*rule{},
+		index: map[string]*rule{},
+		clock: &defaultClock{},
+	}
+
+	for _, records := range rules.SamplingRuleRecords {
+		if records.SamplingRule.RuleName == nil {
+			globalLogger.V(1).Info("Sampling rule without rule name is not supported")
+			failed = true
+			continue
+		}
+
+		// Only sampling rule with version 1 is valid
+		if records.SamplingRule.Version == nil {
+			globalLogger.V(1).Info("Sampling rule without version number is not supported", "RuleName", *records.SamplingRule.RuleName)
+			failed = true
+			continue
+		}
+
+		if *records.SamplingRule.Version != int64(1) {
+			globalLogger.V(1).Info("Sampling rule without version 1 is not supported", "RuleName", *records.SamplingRule.RuleName)
+			failed = true
+			continue
+		}
+
+		if len(records.SamplingRule.Attributes) != 0 {
+			globalLogger.V(1).Info("Sampling rule with non nil Attributes is not applicable", "RuleName", *records.SamplingRule.RuleName)
+			continue
+		}
+
+		if records.SamplingRule.ResourceARN == nil {
+			globalLogger.V(1).Info("Sampling rule without ResourceARN is not applicable", "RuleName", *records.SamplingRule.RuleName)
```

Comment:
These resourceARN checks seem strange, if it's present we should use it in the same way as any other (not restricted to `*`) if not present we should ignore